src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -159,12 +174,19 @@ protected override async Task InitializeAsync(EventContext telemetryContext, Can
             this.coreCount = Environment.ProcessorCount;
 
             this.ValidateParameters();
-
             DependencyPath workloadPackage = await this.packageManager.GetPlatformSpecificPackageAsync(this.PackageName, this.Platform, this.CpuArchitecture, cancellationToken)
                                                     .ConfigureAwait(false);
 
             this.HPLDirectory = workloadPackage.Path;
+            
+            this.loggedInUserName = Environment.GetEnvironmentVariable("SUDO_USER");
+            if (this.loggedInUserName == null)
```

Comment:
have we verified that it returns null if not run with sudo?

Reply:
if we are not sure, i will suggest to check enviroment.username first and if it is sudo then go and check for SUDO_USER bt i will leave on you, how you want to go about it.

And i will also suggest if we can have a common function in vccomponentextensions to determine current user irrespective of linux and windows, you can choose not to and we can do later but if you can that will be very helpful.

Reply:
> have we verified that it returns null if not run with sudo?

Yes, while running unittests which actually run without sudo, it was returning the value of Environment,Username.